app/controllers/api/deploy_groups_controller.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+class Api::DeployGroupsController < Api::BaseController
+  include DeployGroupsCommon
+  skip_before_action :require_project, only: :index
+  before_action :enabled?
+
+  def index
+    render json: paginate(deploy_groups)
+  end
+
+  def project_deploy_groups
+    render json: paginate(stage.deploy_groups)
+  end
+
+  def update_deploy_groups
+    if !production_change?
+      stage.deploy_groups = deploy_groups_to_add
+      stage.save!
+      head :no_content
+    else
+      render json: {message: "Do not modify production via the API"}, status: 403
```

Comment:
seems kind of random / makes the api pretty useless ...

On Tue, Aug 16, 2016 at 12:44 PM, Adam Panzer <notifications@github.com>
wrote:

> In app/controllers/api/deploy_groups_controller.rb
> <https://github.com/zendesk/samson/pull/1198#discussion_r75005943>:
>
> > +
> > +  def index
> > +    render json: paginate(deploy_groups)
> > +  end
> > +
> > +  def project_deploy_groups
> > +    render json: paginate(stage.deploy_groups)
> > +  end
> > +
> > +  def update_deploy_groups
> > +    if !production_change?
> > +      stage.deploy_groups = deploy_groups_to_add
> > +      stage.save!
> > +      head :no_content
> > +    else
> > +      render json: {message: "Do not modify production via the API"}, status: 403
>
> hmm. I guess we could get rid of it. It was an attempt to protect stages
> from modification via the API if they are production.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/zendesk/samson/pull/1198/files/d43bdda94cc05d08fa093f7c694b09953335fe89#r75005943>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAAsZ3DV28_JWK2lyFknvwh600_UdEnQks5qghMQgaJpZM4Jhtog>
> .
>